File: cyanBat/src/main/java/at/smiech/cyanbat/screens/GameOverScreen.java
package at.smiech.cyanbat.screens;

import android.content.Intent;

import java.util.List;

import at.grueneis.game.framework.Game;
import at.grueneis.game.framework.Graphics;
import at.grueneis.game.framework.Input.TouchEvent;
import at.smiech.cyanbat.activities.CyanBatGameActivity;
import at.smiech.cyanbat.activities.MainMenuActivity;

public class GameOverScreen extends CyanBatBaseScreen {

    private List<TouchEvent> touchEvents;
    private Graphics g;

    public GameOverScreen(Game game) {
        super(game);
        init();

    }

    private void init() {
        g = game.getGraphics();
        System.gc();
        initSounds();
    }

    private void initSounds() {
        if (CyanBatGameActivity.gameTrack.isPlaying()) {
            CyanBatGameActivity.gameTrack.stop();
            CyanBatGameActivity.gameTrack.setLooping(false);
        }
    }

    @Override
    public void update(float deltaTime) {
        touchEvents = game.getInput().getTouchEvents();
        for (TouchEvent event : touchEvents) {
            if (event.type == TouchEvent.TOUCH_UP) {
                CyanBatGameActivity.vib.vibrate(250);
                //game.setScreen(new StartScreen(game));
                Intent mainMenuIntent = new Intent(game.getContext(), MainMenuActivity.class);
                game.getContext().startActivity(mainMenuIntent);
            }
        }
        super.update(deltaTime);
    }

    @Override
    public void present(float deltaTime) {
        drawMap(g);
        g.drawPixmap(CyanBatGameActivity.gameOver, 0, 0);
    }

    @Override
    public void pause() {

    }

    @Override
    public void resume() {

    }

    @Override
    public void dispose() {

    }

}

File: cyanBat/src/main/java/at/smiech/cyanbat/screens/GameOverScreen.kt
package at.smiech.cyanbat.screens

import android.content.Intent
import android.graphics.Color

import at.grueneis.game.framework.Game
import at.grueneis.game.framework.Graphics
import at.grueneis.game.framework.Input.TouchEvent
import at.smiech.cyanbat.activities.CyanBatGameActivity
import at.smiech.cyanbat.activities.MainMenuActivity

class GameOverScreen(game: Game) : CyanBatBaseScreen(game) {

    private var g: Graphics? = null

    init {
        init()

    }

    private fun init() {
        g = super.game.graphics
        System.gc()
        initSounds()
    }

    private fun initSounds() {
        if (CyanBatGameActivity.gameTrack.isPlaying) {
            CyanBatGameActivity.gameTrack.stop()
            CyanBatGameActivity.gameTrack.isLooping = false
        }
    }

    override fun update(deltaTime: Float) {
        if (game.input?.touchEvents?.any { it.type == TouchEvent.TOUCH_UP } == true) {
            CyanBatGameActivity.vib.vibrate(250)
            val mainMenuIntent = Intent(super.game.context, MainMenuActivity::class.java)
            super.game.context.startActivity(mainMenuIntent)
        }
        super.update(deltaTime)
    }

    override fun present(deltaTime: Float) {
        g?.clear(Color.BLACK)
        g?.drawPixmap(CyanBatGameActivity.gameOver, 0, 0)
    }

    override fun pause() {

    }

    override fun resume() {

    }

    override fun dispose() {

    }

}
